quantum/observable/fermion/FermionOperator.cpp

```diff
@@ -24,28 +24,31 @@ namespace quantum {
 FermionTerm &FermionTerm::operator*=(const FermionTerm &v) noexcept {
   coeff() *= std::get<0>(v);
 
```

Comment:
- Bug #1
Description:`FermionTerm` would ignore the adjoint of operators whose site were already in the term. For example, the operator 1^ 2^ 2 1 would return 1^ 2^.
Fix: added the else for the second `if`  (lines 45-47).

      if (it->first == site) {
        if (it->second == c_or_a) {
        // zero out this FermionTerm
          ops().clear();
        } else {//this adds the adjoint of operators whose sites are already in the product
          ops().push_back({site, c_or_a});
        }

      } else {
        ops().push_back({site, c_or_a});
      }


- Bug #2
Description: `FermionOperator::operator*=(const FermionOperator &v)` would ignore the identity term
Fix: replaced lines 212-214 by

        if (!newTerms.insert({id, multTerm}).second && kv.first != "I") {
          newTerms.at(id).coeff() += multTerm.coeff();
        }

        if (!newTerms.insert({id, multTerm}).second && kv.first == "I") {
          newTerms.at(id).coeff() = std::get<0>(kv.second) * std::get<0>(vkv.second);
        }

- Bug #3 (not a bug as such)
Description: prevent `toString()` to move all creation operators to the left and annihilation operators to the right. Because I'm using `toString()` in `fromString()` in ADAPT-VQE, this would change my Hamiltonian. Also, it may be misleading because this could be taken as normal ordered operators.